### PR TITLE
Add k8s template files for build automation

### LINKED ISSUE #3752: Template k8s deployment files
Current problem is that during a release build `src/kubernetes/server/server-deployment.yaml` needs to have a version which doesn't yet exist. We should template these files so that when execution a maven build(probably via profile or similar) so that all versions changing in k8s files would get automatically replaced and committed back to git with a tag.

i.e. template file would be `src/templates/kubernetes/server/server-deployment.yaml` with version as a placeholder. When modifying these files we would just need to remember to modify template and then run specific part of a build if wanting to modify snapshots on github. Release build in CI would then do these automatically based on properties set in pom files.

### PR DESCRIPTION
**Pull request by jvalkeal**
- Now autogenerating `src/kubernetes` from `src/templates/kubernetes`
  by replacing property placeholders in those templates files.
- Generation activated by `deploymentfiles` profile during a
  `copy-resources` phase.
- This commit changed skipper-deployment.yaml as version there didn't
  match one from pom when I ran `mvnw process-resources -P deploymentfiles`.
- Fixes #3752

### CODE REVIEW

File: src/kubernetes/skipper/skipper-deployment.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ spec:
     spec:
       containers:
       - name: skipper
-        image: springcloud/spring-cloud-skipper-server:2.2.1.BUILD-SNAPSHOT
+        image: springcloud/spring-cloud-skipper-server:2.3.0.BUILD-SNAPSHOT
```
Comment (ilayaperumalg) [inline src/kubernetes/skipper/skipper-deployment.yaml]: Is it possible to inject the ${skipper.version} defined from the POM file into the template files?

File: src/templates/kubernetes/kafka/kafka-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      app: kafka
+  template:
+    metadata:
+      labels:
+        app: kafka
+        component: kafka-broker
+    spec:
+      containers:
+      - name: kafka
+        image: wurstmeister/kafka:2.12-2.3.0
```
Comment (ilayaperumalg) [inline src/templates/kubernetes/kafka/kafka-deployment.yaml]: At some point of time, we might want to move all these explicit versioning into configuration parameters.

File: src/kubernetes/skipper/skipper-deployment.yaml (hunk 1 above)
Comment (jvalkeal) [reply src/kubernetes/skipper/skipper-deployment.yaml]: That's exactly what happens when this file is generated from a template. This change happened as pom is using skipper 2.3.0.BUILD-SNAPSHOT

File: src/templates/kubernetes/kafka/kafka-deployment.yaml (hunk 2 above)
Comment (jvalkeal) [reply src/templates/kubernetes/kafka/kafka-deployment.yaml]: Yeah true, I just touched those parts which always need a change during a build.

File: src/kubernetes/skipper/skipper-deployment.yaml (hunk 1 above)
Comment (ilayaperumalg) [reply src/kubernetes/skipper/skipper-deployment.yaml]: oh, I see. I get it now.

File: src/templates/kubernetes/kafka/kafka-deployment.yaml (hunk 2 above)
Comment (ilayaperumalg) [reply src/templates/kubernetes/kafka/kafka-deployment.yaml]: ok, we can revisit in an iteration.

**Comment by ilayaperumalg** [comment]
LGTM, merging.

**Pull request merged**